File: src/java/_2849_Determine_if_a_Cell_Is_Reachable_at_a_Given_Time.java
class Solution {
    public boolean isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int horizontal = Math.abs(fx - sx);
        int vertical = Math.abs(fy - sy);
        int need = Math.max(horizontal, vertical);

        if (need > 0)
            return need <= t;
        else
            return t != 1;
    }
}

File: src/python/_2849_Determine_if_a_Cell_Is_Reachable_at_a_Given_Time.py
class Solution:
    def isReachableAtTime(self, sx: int, sy: int, fx: int, fy: int, t: int) -> bool:
        horizontal = abs(fx - sx)
        vertical = abs(fy - sy)
        need = max(horizontal, vertical)

        if need > 0:
            return need <= t
        else:
            return t != 1
